assets/src/components/sharesecret/ShareSecretModal.tsx

```diff
@@ -0,0 +1,230 @@
+import {
+  Button,
+  Code,
+  FormField,
+  Input,
+  Modal,
+  ReturnIcon,
+  Toast,
+} from '@pluralsh/design-system'
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import { useTheme } from 'styled-components'
+import { LayerPositionType } from '@pluralsh/design-system/dist/components/Layer'
+
+import { splitBindings } from '../settings/usermanagement/roles/RoleFormBindings'
+import { BindingInput } from '../utils/BindingInput'
+
+import {
+  PolicyBindingFragment,
+  useShareSecretMutation,
+} from '../../generated/graphql'
+import { isNonNullable } from '../../utils/isNonNullable'
+import { bindingToBindingAttributes } from '../settings/usermanagement/roles/misc'
+
+import { SECRETS_PATH } from '../../routes/secretsRoutesConsts'
+
+import { useShareSecretContext } from './ShareSecretContext'
+
+const getUrl = (handle?: string) =>
+  `https://${window.location.host}/${SECRETS_PATH}/${handle}`
+
+export default function ShareSecretModal() {
+  const theme = useTheme()
+  const { open, setOpen } = useShareSecretContext()
+  const [completed, setCompleted] = useState(false)
+  const [toast, setToast] = useState(false)
+  const [name, setName] = useState('')
+  const [secret, setSecret] = useState('')
+  const [bindings, setBindings] = useState<
+    Pick<PolicyBindingFragment, 'user' | 'group'>[]
+  >([])
+  const disabled = !name || !secret
+  const inputRef = useRef<HTMLInputElement>()
+
+  useEffect(() => inputRef.current?.focus?.(), [])
+
+  // TODO: Reset on each open.
```

Comment:
with the new modal, state is reset automatically when it opens or closes (because the modal will mount/unmount its children). just need the \<Modal> component to wrap around all the logic- have a recent example [here](https://github.com/pluralsh/plural/blob/61088262a6b733f7fc487ad602783b84fbacc141/www/src/components/overview/clusters/plural-cloud/ConsoleInstanceOIDC.tsx#L60-L70)

only caveat is that there's not a great way for this to work when actions need to share state with the main content. in that example I just put it all into one component instead of having the actions separate. the new modal should also focus the first input by default (unless it's prevented with the onOpenAutoFocus prop)

don't have to do it this way but thought it was worth pointing out

Reply:
Thanks for the reminder. I removed the focus handler because as you said it works out of the box. For the reset I kept current logic as otherwise I would need do some more refactoring (action buttons are specified on modal element).

Reply:
Makes sense